utils/atomic_vector.hh

```diff
@@ -62,4 +63,22 @@ public:
             func(_vec[i]);
         }
     }
+
+    // The callback function must not call remove.
+    //
+    // We would take callbacks that take a T&, but we had bugs in the
+    // past with some of those callbacks holding that reference past a
+    // preemption.
+    seastar::future<> for_each_futurized(seastar::noncopyable_function<seastar::future<>(T)> func) {
+        co_await _vec_lock.for_read().lock();
+        auto unlock = seastar::defer([this] {
+            _vec_lock.for_read().unlock();
+        });
```

Comment:
Slightly nicer: `auto holder = co_await _vec_lock.hold_read_lock();`, no need to schedule the unlock manually.